Cleaned up bugs and made gha demo work for docs

AbeOmor opened the pull request:
- Removed the Dev/Prod logic
- Removed Dev references
- Change LowPriorty -> low_priorty for training cluster
- Remove the auto run for infra code
- Remove Terraform login in the infra
- Added dataset step for model training
- Updated to DS2_v2 for online endpoint for quota issues.

sdonohoo reviewed (commented):
This repo was generated from mlops-v2 before the December updates so it looks like most of these updates are to bring it current with the main-dec31 updates in the main repo. Correct?

I see a few issues:
* tf-gha-deploy-infra.yml: set-env-branch removed. This is probably ok but need to ensure get-config gets only the Prod config file. I assume this demo is meant to implement prod only? Before the set-env-branch code, deployments kept defaulting to dev.
* tf-gha-deploy-infra.yml: test-terraform-state-deployment uses the main-dec31 branch of the mlops-templates. These I cleaned out of the main repo before merging that branch. But there may be several more lingering in this demo repo.
* config-infra-prod.yml removes the terraform variables. This won't work as the demo repo uses terraform to deploy infra. No bicep infra is available yet for gha. Working on it.
*

AbeOmor commented:
> This repo was generated from mlops-v2 before the December updates so it looks like most of these updates are to bring it current with the main-dec31 updates in the main repo. Correct?
> 
> I see a few issues:
> 
> * tf-gha-deploy-infra.yml: set-env-branch removed. This is probably ok but need to ensure get-config gets only the Prod config file. I assume this demo is meant to implement prod only? Before the set-env-branch code, deployments kept defaulting to dev.
> * tf-gha-deploy-infra.yml: test-terraform-state-deployment uses the main-dec31 branch of the mlops-templates. These I cleaned out of the main repo before merging that branch. But there may be several more lingering in this demo repo.
> * config-infra-prod.yml removes the terraform variables. This won't work as the demo repo uses terraform to deploy infra. No bicep infra is available yet for gha. Working on it.

I have been following the pattern of the mlops-v2-ado-demo.  So I removed the set-env-branch, also it wasn't working well. 

Note sure about this `test-terraform-state-deployment 1, can you please help me here.

For config-infra-prod.yml, I removed the extra terraform stuff, I thought it was not necessary as it is not in the mlops-v2-ado-demo. I will add it back. Why is it necessary, it seemed to work without it

@sdonohoo
